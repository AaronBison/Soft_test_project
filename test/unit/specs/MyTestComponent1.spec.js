import MyComponent from '@/MyTestComponent1'

import { mount } from '@vue/test-utils'

test('Html_MatchesSnapshot_True', () => {  
  const wrapper = mount(MyComponent)

  expect(wrapper.html()).toMatchSnapshot()
})

test('hasMovie_ContainsItem_True', () => {  
  const wrapper = mount(MyComponent)

  const containsItem = wrapper.vm.hasMovie('Avengers')
  expect(containsItem).toBe(true)
})

test('hasMovie_ContainsItem_False', () => {  
  const wrapper = mount(MyComponent)

  const containsItem = wrapper.vm.hasMovie('Avangers')
  expect(containsItem).toBe(false)
})

test('setData_MatchesSnapshot_True', async () =>{
  const wrapper = mount(MyComponent)

  const movies = wrapper.vm.marvelMovies
  await wrapper.setData({ marvelMovies: [ ...movies, 'Endgame' ]})

  expect(wrapper.html()).toMatchSnapshot()
})

