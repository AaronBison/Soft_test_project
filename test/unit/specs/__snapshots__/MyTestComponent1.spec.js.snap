// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Html_MatchesSnapshot_True 1`] = `
<div>
    <ul>
        <li><strong>Iron Man</strong></li>
        <li><strong>Avengers</strong></li>
        <li><strong>Infitinity War</strong></li>
    </ul>
</div>
`;

exports[`setData_MatchesSnapshot_True 1`] = `
<div>
    <ul>
        <li><strong>Iron Man</strong></li>
        <li><strong>Avengers</strong></li>
        <li><strong>Infitinity War</strong></li>
        <li><strong>Endgame</strong></li>
    </ul>
</div>
`;
